### DISCUSSION TITLE
Can't make Oauth working by using Authentik - Failed to obtain access token

### DISCUSSION THREAD

**Discussion by Wlmrsi** [outline/outline#3960] [Self Hosting]
I am running Authentik as the SSO (Oauth) server with docker installed. 
I used the same setting from [https://github.com/outline/outline/discussions/2716](url) in my Authentik, but received the same message with [https://github.com/outline/outline/discussions/3478](url) as below:

`{"error":"Failed to obtain access token","stack":"InternalOAuthError: Failed to obtain access token\n    
at OAuth2Strategy._createOAuthError (/opt/outline/node_modules/passport-oauth2/lib/strategy.js:423:17)\n    
at /opt/outline/node_modules/passport-oauth2/lib/strategy.js:177:45\n    
at /opt/outline/node_modules/oauth/lib/oauth2.js:191:18\n    
at ClientRequest.<anonymous> (/opt/outline/node_modules/oauth/lib/oauth2.js:162:5)\n    
at ClientRequest.emit (node:events:526:28)\n    at ClientRequest.emit (node:domain:475:12)\n    
at TLSSocket.socketErrorListener (node:_http_client:442:9)\n    
at TLSSocket.emit (node:events:526:28)\n    
at TLSSocket.emit (node:domain:475:12)\n    
at emitErrorNT (node:internal/streams/destroy:157:8)\n    
at emitErrorCloseNT (node:internal/streams/destroy:122:3)\n    
at processTicksAndRejections (node:internal/process/task_queues:83:21)","level":"error","message":"Error during authentication"}
{"label":"http","level":"info","message":"  --> POST /auth/oidc.callback?code=a4c47682783e4756850066fe12917e49&state=b6fe868da691ea98 302 15ms 69b"}
`

Here are the OIDC settings of my .env :

```OIDC_CLIENT_ID=xxxx
OIDC_CLIENT_SECRET=xxxxx
OIDC_AUTH_URI=https://ip:port/application/o/authorize/
OIDC_TOKEN_URI=https://ip:port/application/o/token/
OIDC_USERINFO_URI=https://ip:port/application/o/userinfo/
OIDC_USERNAME_CLAIM=preferred_username
OIDC_DISPLAY_NAME=Authentik
OIDC_SCOPES="openid profile email"
FORCE_HTTPS=false
#ALLOWED_DOMAINS=

**Comment by vgwidt** [comment]
I remember getting some vague errors when setting up Authentik that took some tinkering.  Try removing the " characters from OIDC_SCOPES if the issue is happening when you try to login, make sure your secrets have valid characters (use openssl generator), and make sure the URLs in Providers in Authentik matches exactly what you have in your .env including the https part.

**Reply by Wlmrsi** [reply]
Removed " characters was done, but no different.
Trying now to make outlie HTTPPS work,  but no success yet, if you know the key point with .env, can you help me to check?

``` # –––––––––––––– REQUIRED ––––––––––––––––
NODE_ENV=production
SECRET_KEY=secret code

UTILS_SECRET=secret code

DATABASE_URL=postgres://outline:password@localhost:5432/outline
DATABASE_URL_TEST=postgres://outline:password@localhost:5432/outline_test
DATABASE_CONNECTION_POOL_MIN=
DATABASE_CONNECTION_POOL_MAX=

PGSSLMODE=disable

REDIS_URL=redis://localhost:6379

URL=http://localhost:3000
PORT=3000

AWS_ACCESS_KEY_ID=secret code
AWS_SECRET_ACCESS_KEY=secret code
AWS_REGION=tw-homelab-1
AWS_S3_ACCELERATE_URL=
AWS_S3_UPLOAD_BUCKET_URL=http://localhost:9000
AWS_S3_UPLOAD_BUCKET_NAME=outline
AWS_S3_UPLOAD_MAX_SIZE=26214400
AWS_S3_FORCE_PATH_STYLE=true
AWS_S3_ACL=private

OIDC_CLIENT_ID=secret code
OIDC_CLIENT_SECRET=secret code
OIDC_AUTH_URI=https://localhost:9443/application/o/authorize/
OIDC_TOKEN_URI=https://localhost:9443/application/o/token/
OIDC_USERINFO_URI=https://localhost:9443/application/o/userinfo/

OIDC_USERNAME_CLAIM=preferred_username
OIDC_DISPLAY_NAME=Authentik
OIDC_SCOPES=openid profile email

SSL_KEY=/opt/outline/ssl/outline.key
SSL_CERT=/opt/outline/ssl/outline.crt

FORCE_HTTPS=true
ENABLE_UPDATES=false
WEB_CONCURRENCY=1
MAXIMUM_IMPORT_SIZE=5120000
DEBUG=http


DEFAULT_LANGUAGE=en_US

**Comment by tommoor** [comment]
@Wlmrsi your comment was deleted as it contained secrets.

**Comment by codedninja** [comment]
@Wlmrsi I just finished deploying Outline with Authentik yesterday. The configuration looks exactly like how we have it, might be an issue on Authentik configuration.

Edit: Make sure on Authentik has a signing key set under the provider.

**Reply by Wlmrsi** [reply]
@codedninja 

Thank you so much, may I know do you use HTTP or HTTPS with your outline website?
I am trying to figure out the reason why....
What do you mean of the signing key "Edit: Make sure on Authentik has a signing key set under the provider."?

If it is ok, may I know if my configuration is similar to yours as below? 
![docker_ps](https://user-images.githubusercontent.com/73973812/186063862-c183c351-249c-4c9c-9f10-e091ebf131a0.png)
![1](https://user-images.githubusercontent.com/73973812/186063871-5ea4e486-0f61-4133-9070-cb9009acc2e4.png)
![2](https://user-images.githubusercontent.com/73973812/186063879-d4d6e604-e7b2-4e43-8ad5-decbcbda72a8.png)
![3](https://user-images.githubusercontent.com/73973812/186063897-970689c1-981b-489d-aded-617dc0557f67.png)

I will appreciate your kind help.
Thank you.

**Reply by vgwidt** [reply]
I think your Authentik settings look good but it doesn't line up with your .env settings.
In .env I see you have FORCE_HTTPS enabled but don't look to be using HTTPS for Outline.  Try disabling that.  Also, you have URL set to http://localhost:3000.  Do you have a proxy setup?  You should set that to the URL that you access it from other devices from.

**Reply by Wlmrsi** [reply]
I didn't use a proxy yet, I was referred to some websites such as https://blog.gurucomputing.com.au/doing-more-with-docker/deploying-outline-wiki/
But I have poor knowledge of how to set up the proxy for outline to redirect the URL with port to outline, I tried Nginx proxy server and set up an FQDN but didn't get it to work.
I will try to use a proxy and check again, thanks.
